python/zygote.py

```diff
@@ -74,166 +75,177 @@ def try_dumps(obj, sort_keys=False, allow_nan=False):
-                    args.insert(1, None)
-
-                # call the desired function in the loaded module
-                val = method(*args)
-
-                if fcn == "file":
-                    # if val is None, replace it with empty string
-                    if val is None:
-                        val = ""
-                    # if val is a file-like object, read whatever is inside
-                    if isinstance(val, io.IOBase):
-                        val.seek(0)
-                        val = val.read()
-                    # if val is a string, treat it as utf-8
-                    if isinstance(val, str):
-                        val = bytes(val, "utf-8")
-                    # if this next call does not work, it will throw an error, because
-                    # the thing returned by file() does not have the correct format
-                    val = base64.b64encode(val).decode()
-
-                # Any function that is not 'file' or 'render' will modify 'data' and
-                # should not be returning anything (because 'data' is mutable).
-                if (fcn != "file") and (fcn != "render"):
-                    if val is None:
-                        json_outp = try_dumps(
-                            {"present": True, "val": args[-1]}, allow_nan=False
-                        )
-                    else:
-                        json_outp_passed = try_dumps(
-                            {"present": True, "val": args[-1]},
-                            sort_keys=True,
-                            allow_nan=False,
-                        )
-                        json_outp = try_dumps(
-                            {"present": True, "val": val},
-                            sort_keys=True,
-                            allow_nan=False,
-                        )
-                        if json_outp_passed != json_outp:
-                            sys.stderr.write(
-                                'WARNING: Passed and returned value of "data" differ in the function '
-                                + str(fcn)
-                                + "() in the file "
-                                + str(cwd)
-                                + "/"
-                                + str(file)
-                                + ".py.\n\n passed:\n  "
-                                + str(args[-1])
-                                + "\n\n returned:\n  "
-                                + str(val)
-                                + '\n\nThere is no need to be returning "data" at all (it is mutable, i.e., passed by reference). In future, this code will throw a fatal error. For now, the returned value of "data" was used and the passed value was discarded.'
-                            )
+            data.extend(s.recv(256 * 1024))
+            if len(data) >= 4:
+                if length is None:
+                    length = int.from_bytes(data[0:4], byteorder="big")
+                    data = data[4:]
```

Comment:
Using `del` avoids making a copy. Should do the same in the if case below as well.
```suggestion
                    del data[:4]
```

Reply:
Changed in db4e174.